lib/matplotlib/artist.py

```diff
@@ -898,6 +900,89 @@ def set_zorder(self, level):
         self.pchanged()
         self.stale = True
 
+    def get_top_margin(self):
```

Comment:
> One less thing that we have to support back compatibility with.

I'm not really sold there -- I think it's more important to have everything working consistently than to have some things in the old way and some in the new better way...  Do we already have other things that only exist as properties?